app/services/supplementary-billing/abstraction-billing-period.service.js

```diff
@@ -23,30 +23,22 @@
  * - 1 Jan 2022 to 30 Jun 2022
  * - 1 Jan 2023 to 30 Jun 2023
  *
- * This is why we always return 2 abstraction periods for each charge purpose. For each period, we calculate the
- * 'billable days'. This is how much of the period intersects with the billing period. Finally, we add a 'consider'
- * flag. This is a boolean which identifies whether a period intersects the billing period. For example, if the
- * abstraction period was 1 Jan to 31 Mar, only one of the periods we create would intersect with a billing period of
- * 2022-04-01 to 2023-03-31
- *
- * - 1 Jan 2022 to 31 Mar 2022 - `false`
- * - 1 Jan 2023 to 31 Mar 2023 - `true`
+ * This is why we always calculate 2 abstraction periods for each charge purpose which we then subsequently check and
+ * disgard if they fall outside the billing period. For each valid period we calculate the 'billing period', from
+ * which we calculate the 'billable days'. This is how much of the period intersects with the billing period.
  *
  * @param {Object} billingPeriod Object that has a `startDate` and `endDate` that defines the billing period
  * @param {module:ChargePurposeModel} chargePurpose The abstraction period to be checked
  *
  * @returns {Object[]} An array of abstraction periods
- * @returns {Date} periods[].startDate
- * @returns {Date} periods[].endDate
- * @returns {boolean} periods[].consider
- * @returns {number} periods[].billableDays
  */
 function go (billingPeriod, chargePurpose) {
   const abstractionPeriods = _abstractionPeriods(billingPeriod, chargePurpose)
 
-  _flagPeriodsForConsideration(billingPeriod, abstractionPeriods)
-
-  _calculateBillableDays(abstractionPeriods)
+  for (const abstractionPeriod of abstractionPeriods) {
+    _calculateBillablePeriods(abstractionPeriod, billingPeriod)
```

Comment:
Now we're only looking at an individual period, I think it would be clearer to tweak the name.

```suggestion
    _calculateBillablePeriod(abstractionPeriod, billingPeriod)
```